propagator/propagator.go

```diff
@@ -131,7 +132,27 @@ func (p CloudTraceFormatPropagator) Extract(ctx context.Context, carrier propaga
 		TraceFlags: tf,
 		Remote:     true,
 	}
-	sc := trace.NewSpanContext(scConfig)
+	return trace.NewSpanContext(scConfig), true
+}
+
+// SpanContextFromRequest extracts a trace.SpanContext from the HTTP request req.
+// In this method, SpanID is expected to be stored in big endian.
+func (p CloudTraceFormatPropagator) SpanContextFromRequest(req *http.Request) (sc trace.SpanContext, ok bool) {
```

Comment:
should we have this return an error, so users can debug if they want to?

Extract can print the error (since it can't return it), and we can return it here.

Reply:
@dashpole fair enough. I replaced bool with err.